### PR TITLE
Information about time in speaking

### PR DESCRIPTION
**Pull request by n2oneProgrammer**

### CODE REVIEW

File: modules/server_statisticts/voice_statistics.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            view=View(timeout=None)
+        )
+
+    @command(name="general", description="Get top 10 active users on voice channels", is_hidden=False)
+    async def top10_voice_channels(self, ctx):
+        result = Database().top10_voice_channels(server_id=ctx.guild.id, channel_id=None)
+        await self.generate_top10(ctx, result)
+
+    @command(name="channel", description="get random meme with keyword", is_hidden=False)
+    async def search_meme(self, ctx, channel: discord.VoiceChannel):
```
Comment (jaanonim) [inline modules/server_statisticts/voice_statistics.py]: I don't think name 'search_meme' is right here. 🙃

File: utils/database.py [hunk 2]
```diff
@@ -18,23 +22,33 @@ def _init_(self):
         self.engine = create_engine(os.getenv("DATABASE_URL"), echo=bool(os.getenv("DATABASE_LOG")))
         Base.metadata.create_all(self.engine)
 
-    # Function is only for showing how orm work
-    # TODO: remove later
-    def add_log(self, server_guid: int, text: str):
+    def add_time_on_channel(self, server_id: int, channel_id: int, user_id: int, timeChannel: int):
```
Comment (jaanonim) [inline utils/database.py]: I'm afraid that if we will ad all data base operations in here, this file will be giaiant. 


To do: split database.py into many modules base on functionality (eg. Voice statisic)

File: modules/server_statisticts/voice_channels_statistic.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import time
+
+from utils.database import Database
+
+rows = dict()
```
Comment (jaanonim) [inline modules/server_statisticts/voice_channels_statistic.py]: I don't see use of this variable. (if in use should be class atribute)

File: modules/server_statisticts/voice_statistics.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        await ctx._send(
+            content="",
+            embed=Embed(title="TOP 10 VOICE USERS",
+                        description=top,
+                        color=discord.Color.blue()),
+            view=View(timeout=None)
+        )
+
+    @command(name="general", description="Get top 10 active users on voice channels", is_hidden=False)
+    async def top10_voice_channels(self, ctx):
```
Comment (jaanonim) [inline modules/server_statisticts/voice_statistics.py]: This naming is miss leading. ( for me it mean: "top ten most popular voise channels" not "top ten most active people's on all voice channels")

File: models/connection_time.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+class ConnectionTime(Base):
+    __tablename__ = "connection_times"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    server_id: Mapped[int]
+    channel_id: Mapped[int]
+    user_id: Mapped[int]
+    time: Mapped[int]
+    date: Mapped[int]
```
Comment (jaanonim) [inline models/connection_time.py]: Timestamp or datatime would be better name because date suggest that this is only date without time

**Review by jaanonim — CHANGES_REQUESTED** [review]
Some wrong naming and one wrong code organisation.

**Review by jaanonim — APPROVED** [review]
LGTM

**Review by Dudek03 — APPROVED** [review]

**Pull request merged**